soroban-env-host/src/test/invocation.rs

```diff
@@ -353,3 +353,58 @@ fn error_spoof_rejected() -> Result<(), HostError> {
     assert!(err.is_type(ScErrorType::Context) && err.is_code(ScErrorCode::InvalidAction));
     Ok(())
 }
+
+#[test]
+fn unrecoverable_error_with_cross_contract_try_call() -> Result<(), HostError> {
+    let host = observe_host!(Host::test_host_with_recording_footprint());
+    let contract_id_obj = host.register_test_contract_wasm(ADD_I32);
+    let invoke_contract_id_obj = host.register_test_contract_wasm(INVOKE_CONTRACT);
+
+    let _ = host.clone().test_budget(5789, 10_048_576).enable_model(
+        ContractCostType::WasmInsnExec,
+        6,
+        0,
+        0,
+        0,
+    );
+
+    let a = 4i32;
+    let b = 7i32;
+
+    let res = host.try_call(
+        invoke_contract_id_obj,
+        Symbol::try_from_val(&*host, &"add_with_try")?,
+        test_vec![&*host, a, b, contract_id_obj].into(),
+    );
+
+    // Running out of budget is a unrecoverable error
+    assert!(res.is_err());
```

Comment:
We probably should assert that this is a budget error.

Reply:
Good point. Fixed.